File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        target = a + b
        a = target - b
        b = target - a
        target = (target - b) + (target - a)
        """
        n = nums
        t = target
        d = 0 # difference
        hd = {}
        for i in range(len(n)):
            d = t - n[i]
            b = t - d
            if b in hd:
                return [i, hd[b]]
            else:
                hd[d] = i

File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $arr = array();
        for ($i = 0; $i < count($nums); $i++) {
            $diff = $target - $nums[$i];
            if (array_key_exists($diff, $arr))
                return [$arr[$diff], $i];
            else
                $arr[$nums[$i]] = $i;
        }
    }
}